tests/Cross-chain/Omnichain.ts

```diff
@@ -0,0 +1,425 @@
+import { mine } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+import { ethers, upgrades } from "hardhat";
+import { SignerWithAddress } from "hardhat-deploy-ethers/signers";
+
+import {
+  AccessControlManager,
+  LZEndpointMock,
+  OmnichainExecutorOwner,
+  OmnichainGovernanceExecutor,
+  OmnichainProposalSender,
+  Timelock,
+} from "../../typechain";
+
+describe("OmnichainProposalSender: ", async function () {
+  const localChainId = 1;
+  const remoteChainId = 2;
+  let maxDailyTransactionLimit = 100;
+  let maxDailyReceiveLimit = 100;
+  let proposalId = 1;
+  const delay = 3800;
+  const values = [0];
+  const proposalType = 0;
+  const calldata = ethers.utils.defaultAbiCoder.encode(["uint256"], [delay]);
+  let Payload;
+  let nativeFee;
+
+  let alice: SignerWithAddress,
+    bob: SignerWithAddress,
+    deployer: SignerWithAddress,
+    accessControlManager: AccessControlManager,
+    remoteEndpoint: LZEndpointMock,
+    executorOwner: OmnichainExecutorOwner,
+    executor: OmnichainGovernanceExecutor,
+    remotePath,
+    sender: OmnichainProposalSender,
+    NormalTimelock: Timelock,
+    FasttrackTimelock: Timelock,
+    CriticalTimelock: Timelock,
+    adapterParams,
+    localEndpoint: LZEndpointMock,
+    localPath;
+
+  function makePayload(targets, values, signatures, calldatas, proposalId, proposalType) {
+    const payload = ethers.utils.defaultAbiCoder.encode(
+      ["address[]", "uint256[]", "string[]", "bytes[]", "uint256", "uint8"],
+      [targets, values, signatures, calldatas, proposalId, proposalType],
+    );
+    return payload;
+  }
+
+  async function updateFunctionRegistry(executorOwner) {
+    const functionregistry = [
+      "setOracle(address)",
+      "setMaxDailyReceiveLimit(uint16,uint256)",
+      "pause()",
+      "unpause()",
+      "setConfig(uint16,uint16,uint256,bytes)",
+      "setSendVersion(uint16)",
+      "setReceiveVersion(uint16)",
+      "forceResumeReceive(uint16,bytes)",
+      "setTrustedRemote(uint16,bytes)",
+      "setTrustedRemoteAddress(uint16,bytes)",
+      "setPrecrime(address)",
+      "setMinDstGas(uint16,uint16,uint256)",
+      "setPayloadSizeLimit(uint16,uint256)",
+      "setUseCustomAdapterParams(bool)",
+      "addTimelocks(address[])",
+    ];
+    const removeArray = new Array(functionregistry.length).fill(false);
+    await executorOwner.upsertSignature(functionregistry, removeArray);
+  }
+
+  before(async function () {
+    deployer = (await ethers.getSigners())[0];
+    alice = (await ethers.getSigners())[1];
+    bob = (await ethers.getSigners())[2];
```

Comment:
I would consider naming these as their index or giving them a descriptive name based on the role they will play in the tests, alternatively define them as named accounts in the hardhat config so that user names don't get crossed with the wrong index.

Reply:
I'll update after merge of this PR https://github.com/VenusProtocol/governance-contracts/pull/27